### PR TITLE
add azuredisk support on German Cloud

### LINKED ISSUE #52812: AzureDisk provisioning doesn't use correct endpoint for Azure Germany
**Is this a BUG REPORT or FEATURE REQUEST?**:

> Uncomment only one, leave it on its own line: 
>
/kind bug
> /kind feature

**What happened**:
When creating a Persistent Volume Claim on a cluster hosted in Azure Germany (using ACS-Engine), with the azuredisk provisioner, the wrong endpoint for blob storage is used:
Warning         ProvisioningFailed      Failed to provision volume with StorageClass "default": Put https://redacted.blob.core.windows.net/redacted?restype=container: dial tcp: lookup redacted.blob.core.windows.net on redacted: no such host

The PVC used:
apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  name: database-vc
  labels:
    type: diskstorage
spec:
  accessModes:
  - ReadWriteOnce
  resources:
    requests:
      storage: 100Gi

**What you expected to happen**:
The correct endpoint for Azure Germany is: https://redacted.blob.core.cloudapi.de/

**How to reproduce it (as minimally and precisely as possible)**:
-

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`): 1.7.5
- Cloud provider or hardware configuration**: Azure
- OS (e.g. from /etc/os-release): Ubuntu 16.04

### PR DESCRIPTION
**Pull request by andyzhangx**
**What this PR does / why we need it**:
add azuredisk support on German Cloud, also supports other cloud like China cloud, etc.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #
fixes #52812

@rootfs @Xorecs @wangtt03

**Special notes for your reviewer**:

**Release note**:
need to cherry pick to v1.7, v1.8

### CODE REVIEW

**Comment by andyzhangx** [comment]
@feiskyer

File: pkg/cloudprovider/providers/azure/azure_blobDiskController.go [hunk 1]
```diff
@@ -106,7 +106,7 @@ func (c *BlobDiskController) CreateVolume(name, storageAccount string, storageAc
 		return "", "", 0, err
 	}
```
Comment (liggitt) [inline pkg/cloudprovider/providers/azure/azure_blobDiskController.go]: useHTTPSForBlobBasedDisk?

Comment (andyzhangx) [reply pkg/cloudprovider/providers/azure/azure_blobDiskController.go]: thanks for the check, I should use useHTTPSForBlobBasedDisk

**Comment by rootfs** [comment]
@andyzhangx squash the commits. Since this fixes the regression introduced in 1.7, request a cherry pick PR into 1.7 branch as well.

**Comment by andyzhangx** [comment]
I see there is another PR fixed this bug and already got merged, just close this PR.

**Pull request closed without merging**